### PR TITLE
Internal: Variables - restore variable BE [ED-19631]

### PR DESCRIPTION
**Pull request by mike-elementor**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### ✨ PR Description
Purpose: Enhance the Variables Module REST API to support label and value overrides when restoring deleted variables.

Main changes:
- Added optional label and value parameters to REST API restore endpoint
- Modified Repository.restore() to accept optional property overrides parameter
- Updated existing_variable restoration to apply overrides when provided in request

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using. **[We'd love your feedback!](mailto:product@linearb.io)** 🚀</sub>

### CODE REVIEW

File: tests/phpunit/elementor/modules/variables/classes/test-reset-api.php [hunk 1]
```diff
@@ -261,6 +261,59 @@ public function test_restore_variable() {
 		$this->assertArrayNotHasKey( 'deleted_at', $response_data['data']['variable'] );
 	}
 
+	public function test_restore_variable__with_overrides() {
+		// Arrange
```
Comment (nami-p) [inline tests/phpunit/elementor/modules/variables/classes/test-reset-api.php:265]: I'm not sure if it's necessary, but may be we can add a test case for duplicate label conflicts, wdyt?

Comment (mike-elementor) [reply tests/phpunit/elementor/modules/variables/classes/test-reset-api.php:265]: done

**Review by elchugreeva — APPROVED** [review]

**Review by Batsirai-muchareva — APPROVED** [review]
